python/rascaline/utils.py

```diff
@@ -20,3 +22,225 @@ def _call_with_growing_buffer(callback, initial=1024):
+            "gaussian_sigma_constant", None
+        )
+        new_hypers["max_angular"] = hyperparameters.pop("max_angular", None)
+        new_hypers["max_radial"] = hyperparameters.pop("max_radial", None)
+        new_hypers["cutoff"] = hyperparameters.pop("interaction_cutoff", None)
+
+        if  'radial_basis' in hyperparameters:
+            new_hypers["radial_basis"] = {hyperparameters.pop("radial_basis").title(): {}}
+            if new_hypers["radial_basis"] != "Gto":
+                warnings.warn("WARNING: rascaline currently only supports a Gto basis.")
+
+        if hyperparameters.get("cutoff_function_type", None) == "ShiftedCosine":
+            new_hypers["cutoff_function"] = {
+                hyperparameters.pop("cutoff_function_type", None): {
+                    "width": hyperparameters.pop("cutoff_smooth_width", None)
+                }
+            }
+        else:
+            new_hypers["cutoff_function"] = {"Step": {}}
+            if hyperparameters.get("cutoff_function_type", None) == "RadialScaling":
+                params = hyperparameters.pop("cutoff_function_parameters", None)
+                new_hypers["radial_scaling"] = {
+                    "Willatt2018": {
+                        "exponent": int(params.get("exponent", None)),
+                        "rate": params.get("rate", None),
+                        "scale": params.get("scale", None),
+                    }
+                }
+
+        deprecated_params = [
+            "global_species",
+            "expansion_by_species_method",
+            "soap_type",
+            "compute_gradients",
+        ]
+        if any([d in hyperparameters for d in deprecated_params]):
+            warnings.warn(
+                "{} are not required parameters in the rascaline software infrastructure".format(
+                    ",".join(
+                        [f"`{d}`" for d in deprecated_params if d in hyperparameters]
+                    )
+                )
+            )
+
+        not_supported = [
+            "coefficient_subselection",
+            "covariant_lambda",
+            "gaussian_sigma_type",
+            "inversion_symmetry",
+            "normalize",
+            "optimization_args",
+            "optimization",
+        ]
+        if any([d in hyperparameters for d in not_supported]):
+            warnings.warn(
+                "{} are not currently supported in rascaline".format(
+                    ",".join([f"`{d}`" for d in not_supported if d in hyperparameters])
+                )
+            )
+
+        return {k: v for k,v in new_hypers.items() if v is not None}
+    elif mode == "dscribe":
+        anticipated_hypers = [
+            "rcut",
+            "nmax",
+            "lmax",
+            "species",
+            "sigma",
+            "rbf",
+            "periodic",
+            "crossover",
+            "average",
+            "sparse",
+            "dtype",
```

Comment:
there is also `weighting` which can be translated to either `radial_scaling` (`weighting/function/pow` in dscribe) or `central_atom_weight` (`weighting/w0` in dscribe)